support/azureutil/azureutil.go

```diff
@@ -217,3 +226,64 @@ func VerifyResourceGroupLocationsMatch(ctx context.Context, hc *hyperv1.HostedCl
 
 	return nil
 }
+
+// GetAzureCredentialsFromSecret gets the Service Principal client ID, client secret, and tenant ID from the credentials
+// secret. This function will be modified a bit once the Microsoft sidecar containers support Managed Identity are
+// delivered (expected Oct 2024).
```

Comment:
Do we know when in Oct will happen this to change? Maybe makes sense to wait until the change happens to put the final code?

Reply:
We do not know when we will receive it. Most likely no earlier than the end of October.

I think we should continue on and not hold up the work.